string desc
int64 code
int32 level
